test/user-test.js

```diff
@@ -1 +1,136 @@
+    // it("should be a method", () => {
+    //   expect(user1.viewTrip).to.be.a();
+    // })
+
+    it("should require an id and a date as arguments", () => {
+      expect(agent.viewTrip("2020/04/30")).to.return("Please provide a user id");
+    })
+    // this method should return a trip from the past
+    // should be able to select past trips
+    it("should be able to select trips from the past", () => {
+      let pastTrip = user1.viewTrip(1, "2019/09/16");
+
+      expect(pastTrip.duration).to.equal(8);
+    })
+    // should be able to select present trips
+    it("should be able to select trips a user is currently on", () => {
+      let presentTrip = user7.viewTrip(7, "2020/10/14");
+
+      expect(presentTrip).to.deep.equal({
+        "id":11,
+        "userID":7,
+        "destinationID":3,
+        "travelers":4,
+        "date":"2020/10/14",
+        "duration":4,
+        "status":"approved",
+        "suggestedActivities":[]
+      });
+    })
+    // should be able to select future trips (approved trips)
+    it("should be able to select approved future trips", () => {
+      let futureTrip = user13.viewTrip("2021/02/07");
+
+      expect(futureTrip.destinationID).to.equal(4);
+    })
+    // should be able to select pending trips
+    it("should be able to select pending trips", () => {
+      let pendingTrip = user12.viewTrip("2020/10/17");
+
+      expect(pendingTrip.status).to.equal("pending");
+    })
+  })
+
+  // it should have a method to track spending
+  describe("A method to view spending", () => {
+    // should be able to return past trip spending
+    it("should be able to return the amount spent on a past trip", () => {
+      let tripCost= user5.viewTripCost("2020/04/30");
+
+      expect(tripCost).to.equal(2514);
+    })
+    // (trip.travelers * destination.estimatedFlightCostPerPerson) + (trip.duration * destination.estimatedLodgingCostPerDay)
+    // should be able to return present trip spending
+    it("should be able to return the amount spent on the present trip", () => {
+      let tripCost = user7.viewTripCost("2020/10/14");
+
+      expect(tripCost).to.equal(4320);
+    })
+    // should be able to return future trip spending (approved trips)
+    it("should be able to return the amount spent on approved future trips", () => {
+      let tripCost = user13.viewTripCost("2021/02/07");
+
+      expect(tripCost).to.equal(2360);
+    })
+    // should be able to calculate cost for pending trips
+    it("should be able to return the amount spent on pending trips", () => {
+      let tripCost = user13.viewTripCost("2020/10/17");
+
+      expect(tripCost).to.equal(2820);
+    })
+
+    describe("A method to calculate the agent fee for all trips", () => {
+      // should be able to calculate 10% agent fee for all of the above
+      it("should return the agent fee for any trip", () => {
+        let agentFee1 = viewAgentFee(13, "2020/10/17");
+        let agentFee2 = viewAgentFee(7, "2020/10/14");
```

Comment:
Will viewAgentFee be a method on a class?

Reply:
* Yes. I want this method to work for travelers and agents, so I included it in the User class